Add changelog for release 0.6.0

sbastrakov opened the pull request:
Here is a draft. I started with a [helper script](https://github.com/ComputationalRadiationPhysics/picongpu/blob/dev/docs/propose_changelog.py), which is really nice. Then regrouped PRs and fixed their titles in the log where necessary. That took surprisingly long (or unsurprisingly for @psychocoderHPC i guess).

One thing i don't particularly like is that a lot of those lines are not actually relative to the old release as suggested. E.g. there was a bug introduced to `dev` after the release and later fixed in `dev`, so together they make no change relative to old release. But now i think it would be practically impossible to separate.

The first message before the changes is a bit drafty, but content-wise reflects what i think. The "thanks to" list is same as `creators` in #3923, same order as well.

sbastrakov commented on CHANGELOG.md:
```diff
@@ -1,6 +1,378 @@
+0.6.0
+-----
+
+**Date:** 2021-12-XX
```
Needs to be fixed before releasing (here or in another PR)

sbastrakov commented on CHANGELOG.md:
```diff
@@ -1,6 +1,378 @@
+
+**Date:** 2021-12-XX
+
+C++14, New Solvers, I/O via openPMD API, HIP Support
```
I was doubting how to call those "solvers". Was considering "extended numerics" instead, but thought that was possibly confusing wrt floating-point arithmetic. Also, we internally call those things solvers, however that may be not obvious for everyone

sbastrakov commented on CHANGELOG.md:
```diff
@@ -1,6 +1,378 @@
+We extended PIConGPU with a few new solvers. Binary collisions are now
+available. We added arbitrary-order FDTD Maxwell's solver. All field solvers are
+now compatible with perfectly matched layer absorber, which became default.
+There is a new way of generating incident field based on the total
```
Maybe this should be reduced or excluded altogether? At the release, it's only for Yee (tho the laser is also only for Yee)

psychocoderHPC replied:
Agree with excluding or explicitly say "generating incident filed based  ... for Yee field solver".

sbastrakov replied:
Clarified and shortened

psychocoderHPC commented on CHANGELOG.md:
```diff
@@ -1,6 +1,378 @@
+**User Input Changes:**
+ - Remove HDF5 I/O plugin (replaced with new openPMD plugin) #3361
+ - Remove ADIOS I/O plugin (replaced with new openPMD plugin) #3691
+ - Change default current solver to EmZ #3599
```
We can remove this line, we are on the way to revert to Esirkepov :-( #3932

psychocoderHPC commented on CHANGELOG.md:
```diff
@@ -1,6 +1,378 @@
+   - Update supported CUDA versions #3458
+   - Juwels profile update #3359
+   - Extend reminder to load environment in the docs with instructions for spack #3478
+   - Update license headers for 2021 #3486
```
IMO we can remove this it is definitely not importent for the user

sbastrakov replied:
I think only the last one is truly unimportant. The JUWELS profile could in principle be reused (rather copied-and-adjusted), and the other two are extension to docs. So i would slightly prefer keep the first 3

psychocoderHPC commented on CHANGELOG.md:
```diff
@@ -1,6 +1,378 @@
+   - Juwels profile update #3359
+   - Extend reminder to load environment in the docs with instructions for spack #3478
+   - Update license headers for 2021 #3486
+   - Fix typo in FieldAbsorberTest #3528
```
can be removed too

psychocoderHPC commented on CHANGELOG.md:
```diff
@@ -1,6 +1,378 @@
+   - Extend TBG_macros and help string for changing field absorber #3736
+   - Add link to conda picongpu-analysis-environment file to docs #3738
+   - Add a brief doc page on ways of adding a laser to a simulation #3739
+   - Update cupla to 0.3.0 release #3748
```
can be removed

psychocoderHPC commented on CHANGELOG.md:
```diff
@@ -1,6 +1,378 @@
+   - Update ISAAC plugin documentation #3740
+   - Add a doxygen warning to not remove gamma filter in transition radiation #3857
+   - Extend user documentation with a warning about tools being Linux-only #3462
+ - Update cupla to latest dev #3212
```
can be removed

psychocoderHPC commented on CHANGELOG.md:
```diff
@@ -1,6 +1,378 @@
+   - Add a doxygen warning to not remove gamma filter in transition radiation #3857
+   - Extend user documentation with a warning about tools being Linux-only #3462
+ - Update cupla to latest dev #3212
+ - Update to latest cupla dev #3244
```
can be removed

psychocoderHPC commented on CHANGELOG.md:
```diff
@@ -1,6 +1,378 @@
+ - Update cupla to latest dev #3212
+ - Update to latest cupla dev #3244
+ - Update hemera modules and add openPMDapi module #3270
+ - Provide gitlab CI yaml #3292
```
can be removed

psychocoderHPC commented on CHANGELOG.md:
```diff
@@ -1,6 +1,378 @@
+ - Provide gitlab CI yaml #3292
+ - Activate reduced HZDR CI tests #3282
+ - Set cmake version explicitly for travis CI #3305
+ - HZDR gitlab CI #3308
```
can be removed

psychocoderHPC commented on CHANGELOG.md:
```diff
@@ -1,6 +1,378 @@
+ - Activate reduced HZDR CI tests #3282
+ - Set cmake version explicitly for travis CI #3305
+ - HZDR gitlab CI #3308
+ - Update to cupla 0.3.0-dev #3307
```
can be removed

psychocoderHPC commented on CHANGELOG.md:
```diff
@@ -1,6 +1,378 @@
+ - CI: check code style #3441
+ - CI: more test on HZDR CI #3443
+ - CI: perform branch check in HZDR CI #3449
+ - Update to latest cupla 0.3.0-dev #3454
```
can be removed

psychocoderHPC commented on CHANGELOG.md:
```diff
@@ -1,6 +1,378 @@
+ - Add a new compile-time test suite to test all particle pushers #3386
+ - Use openPMD::getVersion() function #3427
+ - CI: check code style #3441
+ - CI: more test on HZDR CI #3443
```
can be removed

psychocoderHPC commented on CHANGELOG.md:
```diff
@@ -1,6 +1,378 @@
+ - Update to latest cupla with alpaka0.6.0rc4 #3474
+ - Remove openPMD 0.11.0 module for defq #3480
+ - Update summit profile #3492
+ - Update cupla #3501
```
can be removed

psychocoderHPC commented on CHANGELOG.md:
```diff
@@ -1,6 +1,378 @@
+ - Update CI #3509
+ - Update modules in JUWELS profiles #3459
+ - Update Hemera profiles to load the git module #3536
+ - Use HIP 4.0 in CI #3538
```
can be removed

psychocoderHPC commented on CHANGELOG.md:
```diff
@@ -1,6 +1,378 @@
+ - CI: skip compile #3452
+ - CI: new compiler and CUDA #3469
+ - Disable async device streams for CPU backends #3471
+ - CI: latest container v1.1 #3472
```
can be removed

psychocoderHPC reviewed (changes requested):
@sbastrakov Thanks for preparing the changelog. I marked a few PRs that can definitively be removed.

psychocoderHPC commented:
@sbastrakov  could you please finish the changelog that we can merge this pr tomorrow.

psychocoderHPC commented:
@sbastrakov please add #3938 as bugfix to the changelog

sbastrakov commented:
Log updated. I went again through recent backports and added the PRs to the log. Deleted almost all that you suggested, except some documentation PRs, see reply in review.
